triage/views.py

```diff
@@ -162,24 +170,37 @@ class TriageStartPageView(TemplateView):
     template_name = 'triage/start-now.html'
 
 
-class CustomPageView(ArticlesViewedManagerMixin, TemplateView):
-    http_method_names = ['get']
+class CustomPageView(ArticlesViewedManagerMixin, FormView):
     template_name = 'triage/custom-page.html'
+    success_url = reverse_lazy('custom-page')
+    form_class = forms.SectorForm
 
     @cached_property
     def triage_answers(self):
         answer_manager = helpers.TriageAnswersManager(self.request)
         return answer_manager.retrieve_answers()
 
+    def form_valid(self, form):
+        form.update_triage_sector(self.request, self.triage_answers)
+        return super().form_valid(form)
+
     def dispatch(self, request, *args, **kwargs):
         if not self.triage_answers:
             return redirect('triage-start')
+        if self.triage_answers['sector'] is not None:
+            initial_sector = self.triage_answers['sector']
+            self.initial = {'sector': initial_sector}
         return super().dispatch(request, *args, **kwargs)
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['persona'] = forms.get_persona(self.triage_answers)
         context['triage_result'] = self.triage_answers
+        # for people who selected a service category before the sector
+        # question was removed from triage
+        if self.triage_answers[
+                'sector'] and self.triage_answers['sector'].startswith('EB'):
```

Comment:
bit hard to read this. consider something like 

```
context['service_sector'] = self.triage_answers.get('sector', '').startswith('EB')
```

Reply:
that would be nicer but it gives me `'NoneType' object has no attribute 'startswith'`